### ISSUE TITLE
Nuxt 3.5.2 does not break generate when pages have error

### ISSUE THREAD

**Issue by gustavo-iha** [nuxt/nuxt#21304]
### Environment

- Operating System: `Linux`
- Node Version: `v18.14.0`
- Nuxt Version: `3.5.2`
- Nitro Version: `2.4.1`
- Package Manager: `npm@9.3.1`
- Builder:  `vite`
- User Config: `telemetry, ssr, hooks`
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://github.com/gustavoiha/nuxt-3.5.2-generate-errors-reproduction

### Describe the bug

During `npm run generate`, dynamic pages that are set with `nitroConfig.prerender.routes` do not break the generate process if they throw an error.

In the reproduction repo, I added a minimal `/pages/test/[id].vue` that references a `{{ should.break.generate }}` attribute. The error does appear in the generate log as seen in Logs attached below. However, it does not break the operation, which means it sucessfully finishes and a broken page can be deployed.

Is this a bug or intended behavior?

### Additional context

I believe this is related to https://github.com/nuxt/nuxt/issues/14957.

### Logs

```shell-script
✔ Server built in 542ms
✔ Generated public .output/public
ℹ Initializing prerenderer
ℹ Prerendering 4 initial routes with crawler
  ├─ /200.html (19ms)
  ├─ /404.html (1ms)

 ERROR  [unhandledRejection] Cannot read properties of undefined (reading 'should')

  at _sfc_ssrRender (.nuxt/prerender/chunks/app/_nuxt/_id_-f01e050d.mjs:27:72)
  at renderComponentSubTree (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:423:9)
  at renderComponentVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:371:12)
  at renderVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:483:14)
  at renderComponentSubTree (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:438:7)
  at renderComponentVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:371:12)
  at renderVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:483:14)
  at renderVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:487:9)
  at renderComponentSubTree (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:438:7)
  at renderComponentVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:371:12)


 ERROR  [nitro] [dev] [unhandledRejection] TypeError: Cannot read properties of undefined (reading 'should')

  ├─ /test/1 (25ms)

 ERROR  [unhandledRejection] Cannot read properties of undefined (reading 'should')

  at _sfc_ssrRender (.nuxt/prerender/chunks/app/_nuxt/_id_-f01e050d.mjs:27:72)
  at renderComponentSubTree (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:423:9)
  at renderComponentVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:371:12)
  at renderVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:483:14)
  at renderComponentSubTree (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:438:7)
  at renderComponentVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:371:12)
  at renderVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:483:14)
  at renderVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:487:9)
  at renderComponentSubTree (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:438:7)
  at renderComponentVNode (node_modules/@vue/server-renderer/dist/server-renderer.cjs.prod.js:371:12)


 ERROR  [nitro] [dev] [unhandledRejection] TypeError: Cannot read properties of undefined (reading 'should')

  ├─ /test/2 (4ms)
  ├─ /test/1/_payload.json (1ms)
  ├─ /test/2/_payload.json (0ms)
✔ You can now deploy .output/public to any static hosting!
```

**Comment by MedBilelAbidi** [comment]
hi @gustavo-iha  any update or solution/ workaround on this issue as I have the same one.
thanks in advance.

**Comment by danielroe** [comment]
This is resolved in the latest version (3.6.2) which _does_. You can also set `nitro.prerender.failOnError` if you have upgraded nitro to the latest version.
